lightning/src/ln/dual_funding_tests.rs

```diff
@@ -0,0 +1,266 @@
+			},
+			selected_contest_delay: accept_channel_v2_msg.common_fields.to_self_delay,
+		}),
+		holder_pubkeys: ChannelPublicKeys {
+			funding_pubkey: open_channel_v2_msg.common_fields.funding_pubkey,
+			revocation_basepoint: RevocationBasepoint(
+				open_channel_v2_msg.common_fields.revocation_basepoint,
+			),
+			payment_point: open_channel_v2_msg.common_fields.payment_basepoint,
+			delayed_payment_basepoint: DelayedPaymentBasepoint(
+				open_channel_v2_msg.common_fields.delayed_payment_basepoint,
+			),
+			htlc_basepoint: HtlcBasepoint(open_channel_v2_msg.common_fields.htlc_basepoint),
+		},
+		holder_selected_contest_delay: open_channel_v2_msg.common_fields.to_self_delay,
+		is_outbound_from_holder: true,
+		funding_outpoint,
+		channel_type_features,
+	};
+
+	channel
+		.context
+		.get_mut_signer()
+		.as_mut_ecdsa()
+		.unwrap()
+		.provide_channel_parameters(&channel_transaction_parameters);
+
+	let msg_commitment_signed_from_0 = CommitmentSigned {
+		channel_id,
+		signature: channel
+			.context
+			.get_initial_counterparty_commitment_signature_for_test(
+				&&logger_a,
+				channel_transaction_parameters,
+				accept_channel_v2_msg.common_fields.first_per_commitment_point,
+			)
+			.unwrap(),
+		htlc_signatures: vec![],
+		batch: None,
+		#[cfg(taproot)]
+		partial_signature_with_nonce: None,
+	};
+
+	if test_async_persist {
+		chanmon_cfgs[1]
+			.persister
+			.set_update_ret(crate::chain::ChannelMonitorUpdateStatus::InProgress);
+	}
+
+	// Handle the initial commitment_signed exchange. Order is not important here.
+	nodes[1]
+		.node
+		.handle_commitment_signed(nodes[0].node.get_our_node_id(), &msg_commitment_signed_from_0);
+	check_added_monitors(&nodes[1], 1);
+
+	if test_async_persist {
+		let events = nodes[1].node.get_and_clear_pending_events();
+		assert!(events.is_empty());
+
+		chanmon_cfgs[1]
+			.persister
+			.set_update_ret(crate::chain::ChannelMonitorUpdateStatus::Completed);
+		let (outpoint, latest_update, _) = *nodes[1]
+			.chain_monitor
+			.latest_monitor_update_id
+			.lock()
+			.unwrap()
+			.get(&channel_id)
+			.unwrap();
+		nodes[1].chain_monitor.chain_monitor.force_channel_monitor_updated(outpoint, latest_update);
+	}
+
+	let events = nodes[1].node.get_and_clear_pending_events();
+	assert_eq!(events.len(), 1);
+	match events[0] {
+		Event::ChannelPending { channel_id: chan_id, .. } => assert_eq!(chan_id, channel_id),
+		_ => panic!("Unexpected event"),
+	};
+
+	let tx_signatures_msg = get_event_msg!(
+		nodes[1],
+		MessageSendEvent::SendTxSignatures,
+		nodes[0].node.get_our_node_id()
+	);
+
+	assert_eq!(tx_signatures_msg.channel_id, channel_id);
+}
+
+#[test]
+fn test_v2_channel_establishment() {
+	// Only initiator contributes, no persist pending
+	do_test_v2_channel_establishment(
+		V2ChannelEstablishmentTestSession { initiator_input_value_satoshis: 100_000 },
+		false,
+	);
+	// Only initiator contributes, no pending
```

Comment:
s/no pending/persist pending

Reply:
thanks. that was a botched comment of note lol